pdu/Src/tempSensor.c

```diff
@@ -44,13 +59,21 @@ void tempSensorTask(void *pvParameters)
     }
 
     /* TODO: initialize the sensor (write config register if needed) */
+    MX_I2C1_Init();
+    //config register value would be set to 0x28 for continuous measurement, standard functioning
+    HAL_I2C_Mem_Write(&hi2c1,SENSOR_ADDRESS_WRITE,CONFIG_REGISTER,I2C_MEMADD_SIZE_8BIT,&CONFIG_VALUE,1,HAL_MAX_DELAY);
+    
 
     TickType_t xLastWakeTime = xTaskGetTickCount();
+    HAL_Delay(220);
```

Comment:
HAL_delay uses the hardware abstraction layer for our STM32 chips. It is implemented via a NULL polling loop which consumes lot of power and resources compared to using vTaskDelay. vTaskDelay is built for FreeRTOS so always use that.

Also, magic number is not a good practice in coding. replace it with a macro.

Do: `vTaskDelay(pdMS_TO_TICKS(STARTUP_DELAY_MS))`